templates/dex-k8s-authenticator.yaml

```diff
@@ -0,0 +1,40 @@
+apiVersion: kubeaddons.mesosphere.io/v1alpha1
+kind: Addon
+metadata:
+  name: dex-k8s-authenticator
+  namespace: kubeaddons
+spec:
+  kubernetes:
+    minSupportedVersion: v1.14.0
+  chartReference:
+    chart: raw.githubusercontent.com/dex-k8s-authenticator
+    # TODO(mh): If this is going to be used figure out the production URL
+    # See: https://github.com/mintel/dex-k8s-authenticator/issues/70
+    repo: https://raw.githubusercontent.com/mintel/dex-k8s-authenticator/feat-helm-repo/charts/repo/stable
+    version: 1.0.0
+    values: |
+      ---
+      ingress:
+        enabled: true
+        annotations:
+          kubernetes.io/ingress.class: traefik
+        path: /dex-k8s-authenticator
+        hosts:
+          - ""
+      dexK8sAuthenticator:
+        port: 5555
+        debug: false
+        web_path_prefix: /dex-k8s-authenticator
+        #logoUrl: http://<path-to-your-logo.png>
+        #tlsCert: /path/to/dex-client.crt
+        #tlsKey: /path/to/dex-client.key
+        clusters:
+        - name: kubernetes-cluster
+          short_description: "Kubernetes cluster"
+          description: "Kubernetes cluster authenticator"
+          # client_secret: value is generated automatically - see `setupDexK8sAuthenticator()`
+          # This `client_id` must by in sync with `dex.yaml` value as well as
+          # kube-apiserver flag `oidc-client-id`.
+          client_id: kube-apiserver
+          issuer: https://dex-kubeaddons.kubeaddons.svc.cluster.local:8080/dex
```

Comment:
This is an example where this settings cannot work without the dex settings defined below.

Reply:
What would be a preferred way to enforce dependency between `dex` and `dex-k8s-authenticator`?

Reply:
I believe for the moment and in the interest of time we should skip on worrying about this dependency management issue and allow a manual approach in the interim. I have created [https://jira.mesosphere.com/browse/DCOS-56045](DCOS-56045) to cover the next iteration on this wherein we need to automate this.